athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/handlers/GlueMetadataHandler.java

```diff
@@ -259,24 +260,19 @@ public ListSchemasResponse doListSchemaNames(BlockAllocator blockAllocator, List
     protected ListSchemasResponse doListSchemaNames(BlockAllocator blockAllocator, ListSchemasRequest request, DatabaseFilter filter)
             throws Exception
     {
-        GetDatabasesRequest getDatabasesRequest = new GetDatabasesRequest();
-        getDatabasesRequest.setCatalogId(getCatalog(request));
+        GetDatabasesRequest getDatabasesRequest = GetDatabasesRequest.builder()
+                .catalogId(getCatalog(request))
+                .build();
 
         List<String> schemas = new ArrayList<>();
-        String nextToken = null;
-        do {
-            getDatabasesRequest.setNextToken(nextToken);
-            GetDatabasesResult result = awsGlue.getDatabases(getDatabasesRequest);
+        GetDatabasesIterable responses = awsGlue.getDatabasesPaginator(getDatabasesRequest);
 
-            for (Database next : result.getDatabaseList()) {
-                if (filter == null || filter.filter(next)) {
-                    schemas.add(next.getName());
-                }
-            }
-
-            nextToken = result.getNextToken();
-        }
-        while (nextToken != null);
+        responses.stream().forEach(response -> response.databaseList()
+                .forEach(database -> {
```

Comment:
we can simply this to just using java-stream. filter 
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html